src/tests/ftest/util/server_utils.py

```diff
@@ -216,6 +324,236 @@ def create_yaml(self, filename):
+           self.runner.job.yaml_params.is_scm():
+            # Setup the hostlist to pass to dmg command
+            servers_with_ports = [
+                "{}:{}".format(host, self.runner.job.yaml_params.port)
+                for host in self._hosts]
+
+            # Change ownership of attach info file
+            chmod_cmds = [
+            "chmod -R 777 {}/daos_server.attach_info_tmp".format(
+                self.attach.value),
+            ]
+            pcmd(self._hosts, chmod_cmds, False)
+
+            # Format storage and wait for server to change ownership
+            self.log.info("Formatting hosts: <%s>", self._hosts)
+            storage_format(self.daosbinpath, ",".join(servers_with_ports))
+            self.runner.job.mode = "normal"
+            try:
+                self.runner.job.check_subprocess_status(self.runner.process)
+            except CommandFailure as error:
+                self.log.info("Failed to start after format: %s", str(error))
+
+        return True
+
+    def stop(self):
+        """Stop the server through the runner."""
+        self.log.info("Stopping servers")
+        if self.runner.sudo:
+            self.kill()
+            if self.runner.job.yaml_params.is_nvme() or \
+               self.runner.job.yaml_params.is_scm():
+                storage_reset(self._hosts)
+            # Make sure the mount directory belongs to non-root user
+            self.log.info("Changing ownership of mount to non-root user")
+            cmd = "sudo chown -R {0}:{0} /mnt/daos*".format(getpass.getuser())
+            pcmd(self._hosts, cmd, False)
+        else:
+            try:
+                self.runner.stop()
+            except CommandFailure as error:
+                raise ServerFailed("Failed to stop servers:{}".format(error))
+
+    def server_clean(self):
+        """Prepare the hosts before starting daos server."""
+        # Kill any doas servers running on the hosts
+        self.kill()
+        # Clean up any files that exist on the hosts
+        self.clean_files()
+
+    def kill(self):
+        """Forcably kill any daos server processes running on hosts.
+
+        Sometimes stop doesn't get everything.  Really whack everything
+        with this.
+
+        """
+        kill_cmds = [
+            "sudo pkill '(daos_server|daos_io_server)' --signal INT",
+            "sleep 5",
+            "pkill '(daos_server|daos_io_server)' --signal KILL",
+        ]
+        self.log.info("Killing any server processes")
+        pcmd(self._hosts, "; ".join(kill_cmds), False, None, None)
+
+    def clean_files(self):
+        """Clean the tmpfs on the servers."""
+        clean_cmds = [
+            "find /mnt/daos -mindepth 1 -maxdepth 1 -print0 | xargs -0r rm -rf"
+        ]
+
+        if self.runner.job.yaml_params.is_nvme() or \
+           self.runner.job.yaml_params.is_scm():
+            clean_cmds.append("sudo rm -rf /mnt/daos; sudo umount /mnt/daos")
```

Comment:
Should this really be append? Won't the `find /mnt/daos ... xargs -0r rm -ft` fail due to the root file ownership and it should be replaced with the first command in this string (or maybe just have `sudo` added to it)?